spanner/google/cloud/spanner/snapshot.py

```diff
@@ -81,7 +83,10 @@ def read(self, table, columns, keyset, index='', limit=0,
             transaction=transaction, index=index, limit=limit,
             resume_token=resume_token, options=options)
 
-        return StreamedResultSet(iterator)
+        if self._multi_use:
+            return StreamedResultSet(iterator, source=self)
+        else:
+            return StreamedResultSet(iterator)
 
     def execute_sql(self, sql, params=None, param_types=None, query_mode=None,
                     resume_token=b''):
```

Comment:
Thought about this a bit more. I realized that since python client uses the synchronous grpc api, solution I proposed will not work. But the issue still remains. For example I can do this
```
snapshot = Snapshot(multi_use=true)
iter1 = snapshot.read(....)
iter2 = snapshot.read(....)
rows1 = list(iter1)
rows2 = list(iter2)
```

The two reads above will use different transaction ids. I think the easiest solution will be to add an explicit `begin` method which calls BeginTransaction. Users will need to first call that before they can do any reads/queries in case of multi use.